src/Components/Components/test/RendererTest.cs

```diff
@@ -431,12 +433,12 @@ public void CanDispatchEventsToTopLevelComponents()
 
             // Act/Assert: Event can be fired
             var eventArgs = new UIEventArgs();
-            renderer.DispatchEvent(componentId, eventHandlerId, eventArgs);
+            await renderer.DispatchEventAsync(componentId, eventHandlerId, eventArgs);
```

Comment:
I'm not certain we really should be awaiting here, or in any of the other tests. That would be a significant change.

Up until now, we've always guaranteed that the event dispatcher and initial re-render is synchronous. It returns a Task now for error-handling purposes, but that shouldn't change the contract that event dispatching and re-render is synchronous.

I think we should explicitly *not* be awaiting so that we can observe the synchronous effects.

Or if you prefer to await in most tests but just add a few tests to show that the dispatch and rerender actually occur synchronously (including for nested components), that would be fine with me too.

Reply:
In case you're wondering why I think this matters, there are various scenarios in client-side browser code where the behavior is different depending on whether the effects happen synchronously after a user action (such as a click) or if they are async. For example, the "open popup" or "go full screen" APIs are allowed synchronously after a click, but disallowed asynchronously.

Maybe you wouldn't invoke either of those APIs during rendering, but to avoid us having to reason about those fine details on a per-API basis, it's much more reasonable if we guarantee synchrony as much as possible. Asynchrony is hard to reason about and nearly impossible to track down if it starts leaking in.

Reply:
>  I think we should explicitly not be awaiting so that we can observe the synchronous effects.

I think that would require https://github.com/aspnet/AspNetCore/pull/7240/files to go in first